### DISCUSSION TITLE
How to send more than Openssl 3.2.4-dev quicserver.c record layer size

### DISCUSSION THREAD

**Discussion by tonypark7277** [openssl/openssl#25611] [Q&A] [answered]
I build Openssl 3.2.4-dev version on ubuntu 22.04.5 LTS.

To check how QUIC works, I used `quic-client-non-block` with `quicserver` in `util` directory as `README.md` in `demos/guide` says.

Also I changed `quicserver.c` so that it sends 20KB html page.
I also created experimental CA cert and server cert for remote communication.

Then I can check `quicserver` can send upto 8192 bytes before receiving ACK from client.
How can I change the `quicserver.c` code to let server send more data before receiving ACK?

![image](https://github.com/user-attachments/assets/b7b2c971-670f-4fc4-b3fb-68018c3b70b8)

**Comment by nhorman** [comment, marked as answer]
The quicserver.c application isn't really build to send arbitrary amounts of data easily, it send pre-defined responses to the client based on the HTTP 1.0 request it receives.

That said, those responses are defined in the responses array found in the stack frame of the main function.  You can modify those as you see fit to tailor how much data you want to return.

Please note we are currently in the process of writing quic blocking/non-blocking servers here:
https://github.com/openssl/openssl/pull/25431/commits

Which act as an echo server (replying with the same data that was sent) which may give you more control over how much data is returned more easily

**Reply by tonypark7277** [reply]
Thank you for your comment.

However with quic blocking/non-blocking server codes you are working on, 
server can sends **upto 10559 bytes (or upto PKN 10)** before receiving client's ACK.

To be specific, I change quic-server-non-block.c in [https://github.com/openssl/openssl/pull/25431/commits](https://github.com/openssl/openssl/pull/25431/commits), to send dummy 20KB HTTP body with headers instead of echoing client's HTTP request.
Also I changed initcwnd and initrwnd of server machine to 30.

Is there any limitations of QUIC protocol related to this result?  (such as 16384 bytes limitation in TCP/TLS record layor)
